join-chat-to-teams-meeting/client.js

```diff
@@ -10,125 +10,129 @@ let chatThreadClient;
+const messagesContainer = document.getElementById('messages-container');
+const chatBox = document.getElementById('chat-box');
+const sendMessageButton = document.getElementById('send-message');
+const messageBox = document.getElementById('message-box');
 
 var userId = '';
 var messages = '';
 
 async function init() {
-	const connectionString = "<SECRET_CONNECTION_STRING>";
-	const endpointUrl = "<ENDPOINT_URL>";
+  const connectionString =
+    'endpoint=https://acs-ui-dev.communication.azure.com/;accesskey=imlXG4BniVo0S5d0bhPyq4sLT2x9CQN+KlWINIPNkhAFP2QjCXnQiCds6xUutrQ1eYKEVCpYbvjba/OirATWCg==';
+  const endpointUrl = 'https://acs-ui-dev.communication.azure.com';
 
-	const identityClient = new CommunicationIdentityClient(connectionString);
+  const identityClient = new CommunicationIdentityClient(connectionString);
 
-	let identityResponse = await identityClient.createUser();
-	userId = identityResponse.communicationUserId;
-	console.log(`\nCreated an identity with ID: ${identityResponse.communicationUserId}`);
+  let identityResponse = await identityClient.createUser();
+  userId = identityResponse.communicationUserId;
+  console.log(`\nCreated an identity with ID: ${identityResponse.communicationUserId}`);
 
-	let tokenResponse = await identityClient.getToken(identityResponse, [
-		"voip",
-		"chat",
-	]);
+  let tokenResponse = await identityClient.getToken(identityResponse, ['voip', 'chat']);
 
-	const { token, expiresOn } = tokenResponse;
-	console.log(`\nIssued an access token that expires at: ${expiresOn}`);
-	console.log(token);
+  const { token, expiresOn } = tokenResponse;
+  console.log(`\nIssued an access token that expires at: ${expiresOn}`);
+  console.log(token);
 
-	const callClient = new CallClient();
-	const tokenCredential = new AzureCommunicationTokenCredential(token);
-	callAgent = await callClient.createCallAgent(tokenCredential);
-	callButton.disabled = false;
+  const callClient = new CallClient();
+  const tokenCredential = new AzureCommunicationTokenCredential(token);
+  callAgent = await callClient.createCallAgent(tokenCredential);
+  callButton.disabled = false;
 
-	chatClient = new ChatClient(
-		endpointUrl,
-		new AzureCommunicationTokenCredential(token)
-	);
+  chatClient = new ChatClient(endpointUrl, new AzureCommunicationTokenCredential(token));
 
-	console.log('Azure Communication Chat client created!');
+  console.log('Azure Communication Chat client created!');
 }
 
 init();
 
-callButton.addEventListener("click", async () => {
-	// join with meeting link
-	call = callAgent.join({meetingLink: meetingLinkInput.value}, {});
+meetingLinkInput.addEventListener('input', () => {
+  try {
+    // Chat thread Id is included in the meeting link, usually the
+    // part containing the word 'thread'
+    const url = new URL(meetingLinkInput.value);
+    const decodedPath = decodeURIComponent(url.pathname);
+    threadIdInput.value = decodedPath.split('/').find((val) => val.includes('thread'));
```

Comment:
Would probably be better to use something safer, like a regex. According to this [doc](https://learn.microsoft.com/en-us/azure/communication-services/quickstarts/chat/meeting-interop?pivots=platform-web#get-a-teams-meeting-chat-thread-for-a-communication-services-user), there can be more parts with `thread` in it.

This is one I've used in my testing C# app:

```c#
var threadId = WebUtility.UrlDecode(Regex.Match(_configuration.TeamsMeetingLink, "(.*meetup-join\\/)(?<threadId>19.*)(\\/.*)").Groups["threadId"].Value);
```